src/backend/mailbox_manager/admin.py

```diff
@@ -9,6 +9,30 @@
 from mailbox_manager.utils.dimail import DimailAPIClient
 
 
+@admin.action(description=_("Check domain health"))
+def check_domain_health(modeladmin, request, queryset):  # pylint: disable=unused-argument
+    """Admin action to check domain health with dimail and update domain status."""
+    client = DimailAPIClient()
+    domains_updated = []
+    for domain in queryset:
+        old_status = domain.status
+        try:
+            response = client.check_domain(domain)
+        except exceptions.HTTPError as err:
+            messages.error(
+                request,
+                _(f"Check domain health failed for {domain.name} with message: [{err}]"),
+                )
+        else:
+            new_status = domain.status
+            if old_status != new_status:
+                domains_updated.append(domain)
+            messages.success(
+                request,
+                f"Domain updated {','.join(domains_updated)} {response.status_code} {response.json()}"
```

Comment:
This results in an exception when run locally, because `domains_updated` is a list of `MailDomain` objects rather than a list of `str` values…